quickstart-kotlin/src/main/java/com/paypal/checkoutsamples/order/KotlinJetpackComposeQuickStart.kt

```diff
@@ -0,0 +1,194 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Button
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.paypal.checkout.PayPalCheckout
+import com.paypal.checkout.approve.OnApprove
+import com.paypal.checkout.cancel.OnCancel
+import com.paypal.checkout.createorder.CreateOrder
+import com.paypal.checkout.createorder.CurrencyCode
+import com.paypal.checkout.createorder.OrderIntent
+import com.paypal.checkout.createorder.ShippingPreference
+import com.paypal.checkout.createorder.UserAction
+import com.paypal.checkout.error.OnError
+import com.paypal.checkout.order.CaptureOrderResult
+import com.paypal.checkoutsamples.order.usecase.CreateOrderRequest
+import com.paypal.checkoutsamples.order.usecase.CreateOrderUseCase
+import com.paypal.checkoutsamples.ui.theme.AndroidNativeCheckoutSamplesTheme
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+
+class KotlinJetpackComposeQuickStart : ComponentActivity() {
+    private val checkoutState = MutableStateFlow<PaypalCheckoutState>(Loading)
+    private val checkOutSdk:PayPalCheckout
+        get() =  PayPalCheckout
+    private val createdItems = mutableListOf<CreatedItem>()
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        registerPaypalCheckoutCallback(checkOutSdk)
+        val createOrderUseCase by lazy { CreateOrderUseCase() }
+
+        setContent {
+            AndroidNativeCheckoutSamplesTheme {
+                Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colors.background) {
+                    PaymentScreen(paypalSdkInstance = checkOutSdk, orderUseCase = createOrderUseCase,
+                        checkoutState = checkoutState,
+                        createdItems = createdItems)
+                }
+            }
+        }
+    }
+
+    /* represents the different states to be observed when payment is initiated  */
+    sealed interface PaypalCheckoutState
+    object Loading : PaypalCheckoutState
+    object OrderPaidForSuccessfully : PaypalCheckoutState
+    object OrderCapturingFailed : PaypalCheckoutState
+    object OrderPaymentCancelled : PaypalCheckoutState
```

Comment:
I would better make a separate file for this Interface. 
or might be better if we can keep the same callback message that is coming from `PayPal SDK`

Reply:
Haha bet you to it.
I can make a separate file for this interface since relying on the callback message coming from `Paypal Sdk` might not be that flexible because of the tight coupling of the app and `Paypal Sdk` . If the `Paypal Sdk`  callback message change, you'll might have to refactor a lot of code to match the new changes. 
Imo, making your own 'state' interface is better but I am open to guidance from you.

> LGTM.
> 
> I was working on this but I am not able to complete this.
> 
> will try this branch and explore.